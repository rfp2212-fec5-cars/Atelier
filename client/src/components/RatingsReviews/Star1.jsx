import React from 'react';

const Star1 = ({ rate }) => {

  return (
    <div className="star-rating">
      <div className="back-stars">
        <span >☆</span>
        <span >☆</span>
        <span >☆</span>
        <span >☆</span>
        <span >☆</span>

        <div className="front-stars" style={{ width: `${rate * 100 + '%'}` }}>
          <span >★</span>
          <span >★</span>
          <span >★</span>
          <span >★</span>
          <span >★</span>
        </div>
      </div>
    </div>
  );

};

export default Star1;