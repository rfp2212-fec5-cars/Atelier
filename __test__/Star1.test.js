import TestRenderer from 'react-test-renderer';
import Star1 from '../client/src/components/RatingsReviews/Star1.jsx';

const testRenderer = TestRenderer.create(
  <Star1 rate = {0.6}></Star1>
);
describe('star1 testing', ()=>{
  it('should exists', ()=>{
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have 10 span stars', ()=>{
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node=>node.type === 'span')).toHaveLength(10);
  });

});