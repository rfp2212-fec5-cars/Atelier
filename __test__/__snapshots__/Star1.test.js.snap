// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`star1 testing should exists 1`] = `
<div
  className="star-rating"
>
  <div
    className="back-stars"
  >
    <span>
      ☆
    </span>
    <span>
      ☆
    </span>
    <span>
      ☆
    </span>
    <span>
      ☆
    </span>
    <span>
      ☆
    </span>
    <div
      className="front-stars"
      style={
        {
          "width": "60%",
        }
      }
    >
      <span>
        ★
      </span>
      <span>
        ★
      </span>
      <span>
        ★
      </span>
      <span>
        ★
      </span>
      <span>
        ★
      </span>
    </div>
  </div>
</div>
`;
